app/assets/javascripts/discourse/app/models/topic-details.js

```diff
@@ -35,18 +36,23 @@ const TopicDetails = RestModel.extend({
     this.set("loaded", true);
   },
 
-  @discourseComputed("notification_level", "notifications_reason_id")
-  notificationReasonText(level, reason) {
+  @discourseComputed("notification_level", "notifications_reason_id", "topic")
+  notificationReasonText(level, reason, topic) {
     if (typeof level !== "number") {
       level = 1;
     }
 
     let localeString = `topic.notifications.reasons.${level}`;
     if (typeof reason === "number") {
-      const tmp = localeString + "_" + reason;
+      let localeStringWithReason = localeString + "_" + reason;
+
+      if (this._notificationReasonStale(level, reason, User.current(), topic)) {
```

Comment:
I'm trying to move us away from `User.current()` since it's a singleton. Rather than a computed property is there any way this could be a function that's called somewhere like a controller or component where `this.currentUser` is present?

Reply:
Yes I think it can be passed in, will change this.

Reply:
@eviltrout I moved this whole `notificationReasonText` method into the `topic-notification-button` component and moved the tests to that component as well :)